### ISSUE TITLE
Trying out how to open an issue

### ISSUE THREAD

**Issue by Manas2030** [Manas2030/hello-world#2]
I am gonna check if useing triple ticks  ``` works...
Function I tried to train NN to predict values according to function, f(x) = x*sin(3*x)
Here is the code I used to get my observations:
```
using DiffEqFlux, OrdinaryDiffEq, Flux, Optim, Plots

u0 = Float32[0.0]
datasize = 50

tspan = (0.0f0,10.0f0)
tsteps = range(tspan[1],tspan[2],length = datasize)

ode_data = tsteps .* sin.( 3 .* tsteps)

dudt2 = FastChain((x, p) -> x.^3,
                  FastDense(1, 16, tanh),
                  FastDense(16, 1))
prob_neuralode = NeuralODE(dudt2, tspan, Vern7(), saveat = tsteps, abstol=1e-6, reltol=1e-6)

function predict_neuralode(p)
  Array(prob_neuralode(u0, p))
end

# x = number of groups we are dividing the dataset into
function loss_neuralode(p)
           pred = predict_neuralode(p)
           tot_loss = 0
           x = 5
           for i in 1:x
               loss = sum(abs, (ode_data[Int((i-1)*(datasize/x)+1):Int(i*(datasize/x))] .- pred[Int((i-1)*(datasize/x)+1):Int(i*(datasize/x))]))
               tot_loss += loss*loss
           end
           return tot_loss, pred
       end

iter = 0
callback = function (p, l, pred; doplot = true)
  global iter
  iter += 1

  display(l)
  if doplot
    # plot current prediction against data
    plt = scatter(tsteps[:], ode_data[:], label = "data")
    scatter!(plt, tsteps[:], pred[:], label = "prediction")
    display(plot(plt))
  end

  return false
end


result_neuralode = DiffEqFlux.sciml_train(loss_neuralode, prob_neuralode.p, ADAM(0.05), cb = callback, maxiters = 300)

callback(result_neuralode.minimizer,loss_neuralode(result_neuralode.minimizer)...;doplot=true)

function final_loss(p)
           return sum(abs2, (ode_data .- prob_neuralode(u0, p)))
end

print("Final Loss:",final_loss(result_neuralode.minimizer),"\n")
```
Observations:
Training on the whole dataset:
NN Architecture and hyperparameters used:
```
dudt2 = FastChain((x, p) -> x.^3,
                         FastDense(1, 16, tanh),
                         FastDense(16, 1))
prob_neuralode = NeuralODE(dudt2, tspan, Vern7(), saveat = tsteps, abstol=1e-6, reltol=1e-6)
```
Loss function:
```
function loss_neuralode(p)
           pred = predict_neuralode(p)
           loss = sum(abs2, (ode_data .- pred))
           return loss, pred
end
```
Iterations = 300, Loss = 43845.42f0

Taking a bigger NN, that is-
```
dudt2 = FastChain((x, p) -> x.^3,
    FastDense(1, 32, tanh),
    FastDense(32,32,tanh),
    FastDense(32, 1))
```
Iterations = 300, Loss = 43836.41f0
-----------------------------------------
Going back to old NN which is
```
dudt2 = FastChain((x, p) -> x.^3,
            FastDense(1, 16, tanh),
            FastDense(16, 1))
```
And taking a new Loss function (that groups different points):
```
# x = number of groups we are dividing the dataset into
function loss_neuralode(p)
           pred = predict_neuralode(p)
           tot_loss = 0
           x = 5
           for i in 1:x
               loss = sum(abs, (ode_data[Int((i-1)*(datasize/x)+1):Int(i*(datasize/x))] .- pred[Int((i-1)*(datasize/x)+1):Int(i*(datasize/x))]))
               tot_loss += loss*loss
           end
           return tot_loss, pred
       end
```
Observations-
With x = 2, loss = 43907.97f0
With x = 5, loss = 44291.656f0
With x = 10, loss = 43852.062f0

Replacing "tot_loss += loss*loss" with tot_loss += log(cosh(loss))
With x = 2, doesn't work (producing NaN)
With x = 5, loss = 43862.867f0
With x = 10, loss = 43873.414f0
---------------------------------------
Training on the first 30 points of the dataset(datasize = 30), since that will be easier for NN to learn

Training on the whole dataset:
Loss = 5765.231f0

Dividing dataset into groups as before:

With x = 2, loss = 5954.032f0
With x = 5, loss = 5966.1304f0
With x = 10, loss = 6044.2563f0

Again replacing "tot_loss += loss*loss" with tot_loss += log(cosh(loss))

With x = 2, loss = 5941.9297f0
With x = 5, loss = 6614.8916f0
With x = 10, loss = 6424.8003f0
